### ISSUE TITLE
INPC007 should not warn when INPC EventArgs are being cached

### ISSUE THREAD

**Issue by Insire** [DotNetAnalyzers/PropertyChangedAnalyzers#112]
Consider a class implementing ``INotifyPropertyChanged`` which has a method ``OnPropertyChanged`` that raises the ``PropertyChanged`` event. But for various reasons, the EventArgs object is being cache like this
```cs
        private static readonly ConcurrentDictionary<string, PropertyChangedEventArgs> _propertyChangedCache = new ConcurrentDictionary<string, PropertyChangedEventArgs>();
```

```cs
        protected void OnPropertyChanged([CallerMemberName]string propertyName = null)
        {
            var args = _propertyChangedCache.GetOrAdd(propertyName, name => new PropertyChangedEventArgs(propertyName));

            PropertyChanged?.Invoke(this, args);
        }
```

INPC007 will warn on the EventHandler, when it shouldn't.
![grafik](https://user-images.githubusercontent.com/11455827/60766949-d2a18600-a0b0-11e9-963e-4f533b3baa87.png)

**Comment by Insire** [comment]
You can find a realworld scenario [here](https://github.com/Insire/MvvmScarletToolkit/blob/master/MvvmScarletToolkit.Observables/ViewModels/ViewModelListBase.cs). You need to update the [ruleset](https://github.com/Insire/MvvmScarletToolkit/blob/master/MvvmScarletToolkit.ruleset) for the warning to appear.

**Comment by JohanLarsson** [comment]
Thanks for reporting!

Just FYI you want:
```cs
        protected void OnPropertyChanged([CallerMemberName]string propertyName = null)
        {
            var args = _propertyChangedCache.GetOrAdd(propertyName ?? string.Empty, name => new PropertyChangedEventArgs(propertyName));

            PropertyChanged?.Invoke(this, args);
        }
```
